code/modules/reagents/reagent_containers.dm

```diff
@@ -15,20 +15,42 @@
 	var/temperature_min = 0 // To limit the temperature of a reagent container can atain when exposed to heat/cold
 	var/temperature_max = 10000
 
+/obj/item/reagent_containers/proc/can_set_transfer_amount(mob/user)
+	if(!length(possible_transfer_amounts))
+		// Nothing to configure.
+		return FALSE
+	if(isrobot(user) && src.loc == user)
+		// Borgs can configure their modules.
+		return TRUE
+	if(!Adjacent(user))
+		// No configuring the beaker across the room.
+		return FALSE
+	if(!ishuman(user))
+		// Although a funny idea, station pets changing transfer
+		// amounts on random conatiners would probably be frustrating
+		// for the crew.
+		return FALSE
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		// I guess there's, like, a switch or a dial or something?
+		// Whatever, you need to use your hands for this.
+		return FALSE
+	return TRUE
+
 /obj/item/reagent_containers/AltClick(mob/user)
-	if(!Adjacent(user) || !length(possible_transfer_amounts) || !ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+	if(!can_set_transfer_amount(user))
 		return
 
 	var/new_transfer_rate = tgui_input_list(user, "Amount per transfer from this:", "[src]", possible_transfer_amounts, "[amount_per_transfer_from_this]")
 	if(!new_transfer_rate)
 		return
 
-	if(!Adjacent(user))
-		to_chat(user, "<span class='warning'>You have moved too far away!</span>")
-		return
-	if(!ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
-		to_chat(user, "<span class='warning'>You can't use your hands!</span>")
-		return
+	if(!can_set_transfer_amount(user))
+		if(!Adjacent(user))
+			to_chat(user, "<span class='warning'>You have moved too far away!</span>")
+			return
+		if(!ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+			to_chat(user, "<span class='warning'>You can't use your hands!</span>")
+			return
```

Comment:
```suggestion
		return
```
Should be indented in this scope then. Else if `can_set_transfer_amount` returns false but you didn't clone all checks here, you would still continue.